src/octree/batch_iterator.rs

```diff
@@ -153,14 +136,11 @@ impl<'a> BatchIterator<'a> {
     {
         let mut point_stream =
             PointStream::new(self.batch_size, self.local_from_global.clone(), &mut func);
-        let mut iterator: Box<Iterator<Item = Point>> = match &self.culling {
-            PointCulling::Any() => Box::new(self.octree.all_points()),
-            PointCulling::Aabb(aabb) => Box::new(self.octree.points_in_box(aabb)),
-            PointCulling::Obb(obb) => Box::new(self.octree.points_in_obb(obb)),
-            PointCulling::Frustum(frustum) => Box::new(self.octree.points_in_frustum(frustum)),
-            PointCulling::OrientedBeam(beam) => Box::new(self.octree.points_in_oriented_beam(beam)),
-        };
-        iterator.try_for_each(|point: Point| point_stream.push_point_and_callback(point))?;
-        point_stream.callback()
+        let mut iterator: Box<Iterator<Item = NodeId>> =
+            get_node_id_iterator(&self.octree, &self.culling);
```

Comment:
here the origin of most changes. (still wip)

Reply:
Where does get_node_id_iterator come from? I don't see it in the other PR.